src/daemons/MetaDaemon.cpp

```diff
@@ -95,6 +86,25 @@ int main(int argc, char *argv[]) {
     }
     auto& localhost = hostAddrRet.value();
 
+    // folly IOThreadPoolExecutor
+    auto ioPool = std::make_shared<folly::IOThreadPoolExecutor>(FLAGS_num_io_threads);
+
+    LOG(INFO) << "Starting Meta HTTP Service";
+    nebula::WebService::registerHandler("/meta", [&localhost, &ioPool] {
```

Comment:
The second parameter of `registerHandler` is a handler generator, which MIGHT not be invoked immediately in this very context. On the other hand, we shoud never assume that a callback is called in a synchronous way, even it truely is.

The problem here is that you are capturing local objects by references, while `WebService::start` might be moved to another context.

Be careful about **context**.